application/packages/python/ftrs_data_layer/models.py

```diff
@@ -170,6 +171,71 @@ class Telecom(BaseModel):
     web: str | None
 
 
+class OpeningTimeCategory(str, Enum):
+    availableTime = "availableTime"
+    availableTimeVariations = "availableTimeVariations"
+    availableTimePublicHolidays = "availableTimePublicHolidays"
+    notAvailable = "notAvailable"
+
+
+class DayOfWeek(str, Enum):
+    monday = "mon"
+    tuesday = "tue"
+    wednesday = "wed"
+    thursday = "thu"
+    friday = "fri"
+    saturday = "sat"
+    sunday = "sun"
+
+
+class AvailableTime(BaseModel):
+    id: UUID = uuid4()
+    category: Literal[OpeningTimeCategory.availableTime] = (
+        OpeningTimeCategory.availableTime
+    )
+    dayOfWeek: DayOfWeek
+    startTime: time
+    endTime: time
+    allDay: bool = False
+
+
+class AvailableTimeVariation(BaseModel):
+    id: UUID = uuid4()
+    category: Literal[OpeningTimeCategory.availableTimeVariations] = (
+        OpeningTimeCategory.availableTimeVariations
+    )
+    description: str
+    startTime: datetime
+    endTime: datetime
+
+
+class AvailableTimePublicHolidays(BaseModel):
+    id: UUID = uuid4()
+    category: Literal[OpeningTimeCategory.availableTimePublicHolidays] = (
+        OpeningTimeCategory.availableTimePublicHolidays
+    )
+    startTime: time
+    endTime: time
+
+
+class NotAvailable(BaseModel):
+    id: UUID = uuid4()
+    category: Literal[OpeningTimeCategory.notAvailable] = (
+        OpeningTimeCategory.notAvailable
+    )
+    description: str
+    startTime: date
+    allDay: bool = True
+
```

Comment:
```suggestion
class BaseOpeningTime(BaseModel):
    """
    Base model for opening times.
    """
    id: UUID = uuid4()
    category: OpeningTimeCategory
    startTime: datetime
    endTime: datetime


class AvailableTime(BaseOpeningTime):
    category: Literal[OpeningTimeCategory.availableTime] = OpeningTimeCategory.availableTime
    dayOfWeek: DayOfWeek
    allDay: bool = False


class AvailableTimeVariation(BaseOpeningTime):
    category: Literal[OpeningTimeCategory.availableTimeVariations] = OpeningTimeCategory.availableTimeVariations
    description: str


class AvailableTimePublicHolidays(BaseOpeningTime):
    category: Literal[OpeningTimeCategory.availableTimePublicHolidays] = OpeningTimeCategory.availableTimePublicHolidays
    startTime: time
    endTime: time


class NotAvailable(BaseOpeningTime):
    category: Literal[OpeningTimeCategory.notAvailable] = OpeningTimeCategory.notAvailable
    description: str
    allDay: bool = True
```
Might be good create a baseOpeningTime model so we avoid the duplication

Reply:
Originally I was going to but it made the json formatting quite ugly with how it's serialising right now.
Also only id and category are shared across all the items in the same format and conditions, so we would still need to manage most these fields in the children.